deps.ffmpeg/99-ffmpeg.zsh

```diff
@@ -158,7 +158,7 @@ config() {
       )
 
       if (( ! shared_libs )) {
-        ff_ldflags+=(-Wl,-Bstatic -pthread)
+        ff_ldflags+=(-Wl,-Bstatic -pthread -Wl,-Bdynamic)
```

Comment:
If `shared_libs` is explicitly disabled, do not build dynamic libraries. This setting goes against the explicit user choice.

Reply:
No, it should not be removed. 

https://github.com/mstorsjo/llvm-mingw/issues/311#issuecomment-1312589456

This gonna be a long read, but the TLDR is the default state of the linker args for external libs without `-Wl,-Bstatic` is dynamic. By not switching it back to dynamic causes problems, at least on llvm toolchain. Regardless of the toolchain used, this should be rigorous.

Reply:
As that is only necessary for the `llvm`-based toolchain, it should be limited to that and not removed for other architectures.